include/mxnet/ndarray.h

```diff
@@ -611,6 +565,64 @@ class NDArray {
              << "CheckAndAllocAuxData is not intended for kDefaultStorage";
     ptr_->CheckAndAllocAuxData(i, aux_shape);
   }
+
+#if MXNET_USE_MKLDNN == 1
+  bool IsMKLDNN() const {
+    return ptr_->IsMKLDNN();
+  }
+  bool IsDefault() const {
```

Comment:
Default is different from defaultstorage. The name is confusing.

How about IsVectorized()?

Also, move this function out of `#if MXNET_USE_MKLDNN` to avoid excessive switches across the codebase.